File: HackerRank Solutions/Algorithms/Implementation/Between Two Sets.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;
import java.util.regex.*;
import java.util.stream.*;
import static java.util.stream.Collectors.joining;
import static java.util.stream.Collectors.toList;

class Result {

    /*
     * Complete the 'getTotalX' function below.
     *
     * The function is expected to return an INTEGER.
     * The function accepts following parameters:
     *  1. INTEGER_ARRAY a
     *  2. INTEGER_ARRAY b
     */

    private static int gcd(int a, int b) {
        if (b == 0) {
            return a;
        }
        return gcd(b, a % b);
    }

    private static int gcd(List<Integer> X) {
        int g = X.get(0);
        for (int i = 1; i < X.size(); i++) {
            g = gcd(g, X.get(i));
        }
        return g;
    }

    private static int lcm(int a, int b) {
        return (a * b) / gcd(a, b);
    }

    private static int lcm(List<Integer> X) {
        int l = X.get(0);
        for (int i = 1; i < X.size(); i++) {
            l = lcm(l, X.get(i));
        }
        return l;
    }

    public static int getTotalX(List<Integer> a, List<Integer> b) {
        int l = lcm(a);
        int g = gcd(b);
        int count = 0;
        for (int i = l, j = 2; i <= g; i = l*j, j++) {
            if (g % i == 0) {
                count++;
            }
        }
        return count;
    }

}

public class Solution {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        String[] firstMultipleInput = bufferedReader.readLine().replaceAll("\\s+$", "").split(" ");

        int n = Integer.parseInt(firstMultipleInput[0]);

        int m = Integer.parseInt(firstMultipleInput[1]);

        List<Integer> arr = Stream.of(bufferedReader.readLine().replaceAll("\\s+$", "").split(" "))
            .map(Integer::parseInt)
            .collect(toList());

        List<Integer> brr = Stream.of(bufferedReader.readLine().replaceAll("\\s+$", "").split(" "))
            .map(Integer::parseInt)
            .collect(toList());

        int total = Result.getTotalX(arr, brr);

        bufferedWriter.write(String.valueOf(total));
        bufferedWriter.newLine();

        bufferedReader.close();
        bufferedWriter.close();
    }
}

File: HackerRank Solutions/Algorithms/Implementation/Between Two Sets.py
#!/bin/python3

import math
import os
import random
import re
import sys

#
# Complete the 'getTotalX' function below.
#
# The function is expected to return an INTEGER.
# The function accepts following parameters:
#  1. INTEGER_ARRAY a
#  2. INTEGER_ARRAY b
#

def gcd(a, b):
    if (b == 0):
        return a
    return gcd(b, a%b)

def gcdOfList(X):
    g = X[0]
    for i in range(1, len(X)):
        g = gcd(g, X[i])
    return g

def lcm(a, b):
    return (a * b) / gcd(a, b)

def lcmOfList(X):
    l = X[0]
    for i in range(1, len(X)):
        l = lcm(l, X[i])
    return l

def getTotalX(a, b):
    l = lcmOfList(a)
    g = gcdOfList(b)
    count = 0
    i = l
    j = 2
    while(i <= g):
        if (g % i == 0):
            count += 1
        i = l*j
        j += 1
    return count

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    first_multiple_input = input().rstrip().split()

    n = int(first_multiple_input[0])

    m = int(first_multiple_input[1])

    arr = list(map(int, input().rstrip().split()))

    brr = list(map(int, input().rstrip().split()))

    total = getTotalX(arr, brr)

    fptr.write(str(total) + '\n')

    fptr.close()
